### PR TITLE
API doc for perpetual account lockout prevention

### PR DESCRIPTION
**Pull request by yumingcao-okta**
## Description:
- API doc for perpetual account lockout prevention
- Monolith release: Yes, 2023.09.0

### Resolves:

* [OKTA-553106](https://oktainc.atlassian.net/browse/OKTA-553106)

### Primary reviewers:
@vanngo-okta @okta/mfaplatform-team

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 1]
```diff
@@ -773,6 +773,7 @@ Fetch a user by `id`, `login`, or `login shortname` if the short name is unambig
 | Parameter | Description                                                        | Param Type | DataType | Required |
 | --------- | ------------------------------------------------------------------ | ---------- | -------- | -------- |
 | id        | `id`, `login`, or `login shortname` (as long as it is unambiguous) | URL        | String   | TRUE     |
+| expand        | Valid value: `block`. If you have migrated to Okta Identity Engine, you can specify `blocks` to include account block details in the `_embedded` attribute. The embedded object will be the same as the one returned by the [Get User's Blocks](#get-user-s-blocks) operation | Query        | String   | FALSE     |
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:776]:
```suggestion
| expand  <ApiLifecycle access="ie" />  | Valid value: `block`. If this parameter is specified, then account block details are included in the `_embedded` attribute. The returned embedded object is the same as the one returned by the [Get User Blocks](#get-user-blocks) operation. | Query        | String   | FALSE     |
```

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 2]
```diff
@@ -2135,6 +2136,47 @@ curl -v -X GET \
 ]
 ```
 
+### Get User's Blocks
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:2139]:
```suggestion
### Get User Blocks
```

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 3]
```diff
@@ -2135,6 +2136,47 @@ curl -v -X GET \
...
+<ApiOperation method="get" url="/api/v1/users/${userId}/blocks" />
+
+Lists information about how the account is blocked from access
+
+##### Request parameters
+
+
+| Parameter | Description  | Param Type | DataType | Required |
+| --------- | ------------ | ---------- | -------- | -------- |
+| id        | `id`, `login`, or `login shortname` (as long as it is unambiguous) of user | URL        | String   | TRUE     |
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:2150]:
```suggestion
| id        | `id`, `login`, or `login shortname` (as long as it is unambiguous) of the user | URL        | String   | TRUE     |
```

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 4]
```diff
@@ -4313,6 +4354,16 @@ Here are some links that may be available on a User, as determined by your polic
 }
 ```
 
+#### User Block object
+
+The object describes how the account is blocked from access. If `appliesTo` is `ANY_DEVICES`, the account is blocked for all devices. If `appliesTo` is `UNKNOWN_DEVICES`, the account is only [blocked for unknown devices](/docs/reference/api/attack-protection/#user-lockout-settings-object).
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:4359]:
```suggestion
The User Block object describes how the account is blocked from access. If `appliesTo` is `ANY_DEVICES`, then the account is blocked for all devices. If `appliesTo` is `UNKNOWN_DEVICES`, then the account is only [blocked for unknown devices](/docs/reference/api/attack-protection/#user-lockout-settings-object).
```

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 5]
```diff
@@ -4313,6 +4354,16 @@ Here are some links that may be available on a User, as determined by your polic
...
+
+| Property  | Description | Datatype |
+| :---------| :---------- | :------- |
+| type      | Type of the block. Valid values: `DEVICE_BASED` | String |
```
Comment (vanngo-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:4363]:
```suggestion
| type      | Type of the block. Valid value: `DEVICE_BASED` | String |
```

**Review by vanngo-okta — CHANGES_REQUESTED** [review]
Looks good! Just a few suggestions.

**Comment by yumingcao-okta** [comment]
@vanngo-okta addressed your comments (for changes to existing API as we discussed), please re-review, thanks!

**Review by vanngo-okta — APPROVED** [review]
LGTM! 🚀

**Comment by vanngo-okta** [comment]
Hi @yumingcao-okta, I'm putting this PR on hold until the feature releases. Please let us know. Thanks!

**Pull request merged**